### PR TITLE
Define "Manifest" and use it to add return types to the "Fetch Manifest" algorithm

### PR DESCRIPTION
**Pull request by samuelgoto**
Start adding return types to algorithms to make them more precise. Will add more as I go along.

***
<a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/pull/271.html" title="Last updated on Jun 8, 2022, 3:54 PM UTC (c654f3c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/271/1ccd28d...c654f3c.html" title="Last updated on Jun 8, 2022, 3:54 PM UTC (c654f3c)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -1258,7 +1260,7 @@ to keep their actual manifest on an arbitary path while allowing the user agent
 manipulation to fingerprint. See https://github.com/fedidcg/FedCM/issues/230 for more details on
 this attack.
 
-To <dfn>check the root manifest</dfn> given a {{FederatedIdentityProvider}} |provider|:
+The <dfn>check the root manifest</dfn> accepts a {{FederatedIdentityProvider}} |provider| and returns true if the manifest is included in the manifest list and false otherwise:
```
Comment (npm1) [inline spec/index.bs:1263]:
```suggestion
The <dfn>check the root manifest</dfn> algorithm accepts a {{FederatedIdentityProvider}} |provider| and returns whether the manifest is included in the manifest list:
```

File: spec/index.bs [hunk 2]
```diff
@@ -1281,15 +1283,15 @@ To <dfn>check the root manifest</dfn> given a {{FederatedIdentityProvider}} |pro
...
+The <dfn>fetch the internal manifest</dfn> algorithm accepts a {{FederatedIdentityProvider}} |provider| and returns a [=Manifest=]:
     1. Run a [[!CSP]] check with a [[CSP#directive-connect-src|connect-src]]
         directive on the URL passed as |provider|'s {{FederatedIdentityProvider/url}}
         and return if the check fails.
     1. Let the |manifest url| be the file "fedcm.json" relative to the directory
         prefix passed as |provider|'s {{FederatedIdentityProvider/url}}.
-    1. Return the result of fetching the |manifest url| with the
-        [[#Sec-FedCM-CSRF|Sec-FedCM-CSRF]] header but without the
-        [=Identity Provider=]'s cookies.  This request MUST NOT follow
+    1. Return the result of fetching the |manifest url| and parsing it into a [=Manifest=].
```
Comment (npm1) [inline spec/index.bs:1292]: This step is very handwavy: the fetch and parsing are actually a bunch of steps. Add a Note / TODO?

**Review by npm1 — APPROVED** [review]
LGTM % comments

File: spec/index.bs (hunk 1 above)
Comment (samuelgoto) [reply spec/index.bs:1263]: ah, yeah, much better! thanks, using it!

File: spec/index.bs (hunk 2 above)
Comment (samuelgoto) [reply spec/index.bs:1292]: Ah, yes, good point. Will make that clearer in a follow up PR.

Comment (samuelgoto) [reply spec/index.bs:1292]: Added a TODO for this PR.

**Pull request merged**